src/Aspire.Dashboard/Model/ThemeManager.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Dashboard.Model;
+
+public sealed class ThemeManager
+{
+    private readonly object _lock = new object();
+    private readonly List<ModelSubscription> _subscriptions = new List<ModelSubscription>();
+
+    // Note: This won't have a valid value until it has been changed.
+    // If there is a reason to get the theme before changes then the ThemeManager will need to be improved.
```

Comment:
Should you make this into a doc comment so it shows up in intellisense when someone tries to use it in the future? I tend to think we don't need to rework it to fix this now since we don't have a use case, but if we could do something additional to help folks not trip over it in the future I think that'd be good.